src/storage/src/render.rs

```diff
@@ -489,3 +490,39 @@ pub fn build_export_dataflow<A: Allocate>(
         })
     });
 }
+
+pub(crate) fn build_oneshot_ingestion_dataflow<A: Allocate>(
+    timely_worker: &mut TimelyWorker<A>,
+    storage_state: &mut StorageState,
+    ingestion_id: GlobalId,
+    collection_id: GlobalId,
+    collection_meta: CollectionMetadata,
+    description: OneshotIngestionRequest,
+) {
+    let (results_tx, results_rx) = tokio::sync::mpsc::unbounded_channel();
+    let callback = move |result| {
+        // TODO(parkmycar) Do we care if the receiver has gone away?
+        // How do we handle ProtoBatch cleanup if one is sitting in the channel
+        // but the receiver has gone away?
```

Comment:
Since every process can crash whenever, it seems like we need some form of persist garbage collection to guarantee that batches are always cleaned up. If that's the case then maybe we don't have to worry about cleaning up too much here.

Reply:
We are working on that exact thing in Persist now! Re-worked this comment to more accurately describe the situation